sys-devel/gcc_bootstrap/gcc_bootstrap-11.2.0_2021_07_28.recipe

```diff
@@ -232,48 +234,61 @@ BUILD()
 		"../config.h"
 	sed -i -e 's,#define _GLIBCXX_HAVE_TLS 1,/* #undef _GLIBCXX_HAVE_TLS */,' \
 		"../include/$effectiveTargetMachineTriple/bits/c++config.h"
-	make CFLAGS="-O2 $kernelCcFlags" CXXFLAGS="-O2 $kernelCcFlags"
+	make $jobArgs CFLAGS="-O2 $kernelCcFlags" CXXFLAGS="-O2 $kernelCcFlags"
 	mv .libs/libsupc++.a .libs/libsupc++-kernel.a
 	mv saved/libsupc++* .libs/
 	mv saved/gthr-default.h "../include/$effectiveTargetMachineTriple/bits/"
 	mv saved/config.h ..
 	mv saved/c++config.h "../include/$effectiveTargetMachineTriple/bits/"
 	rmdir saved
 
+
 	## BOOTLOADER libgcc + libsupc++
 	local bootCcFlags
 	if [ $effectiveTargetArchitecture == arm ]; then
 		# EFI arm (32-bit) requires software fp
-		bootCcFlags+="-mfloat-abi=soft";
+		bootCcFlags+="-mfloat-abi=soft -nostartfiles -fshort-wchar";
```

Comment:
does bootloader libgcc + libsupc++ on other architectures also potentially need -nostartfiles ?

I could see us looking for these bootloader libgcc / libsupc++ libraries generically on all architectures instead of only on arm.

Reply:
nostartfiles tells gcc not to link against crti.o, start_dyn.o, init_term_dyn.o
These are in fact not needed for building the bootloader so I think generally it would make sense for all architectures.

short-wchar tells gcc to use 2-byte wchar_t (default seems to be 4-byte wchar_t)
This was added for better compatibility with EFI call conventions. Even though I don't see wchar_t being used in the bootloader at all so it's more like to silence warnings.